tesseract_kinematics/core/include/tesseract_kinematics/core/utils.h

```diff
@@ -330,18 +334,9 @@ inline Manipulability calcManipulability(const Eigen::Ref<const Eigen::MatrixXd>
       data.measure = std::sqrt(data.condition);
     }
 
-    data.volume = 1;
-    for (Eigen::Index i = 0; i < sm.eigenvalues().size(); ++i)
-    {
-      // If an eigen value is approximately zero set the volume to zero and return
-      if (tesseract_common::almostEqualRelativeAndAbs(data.eigen_values[i], 0))
-      {
-        data.volume = +0;
-        break;
-      }
-      data.volume = data.volume * data.eigen_values[i];
-    }
-    data.volume = std::sqrt(data.volume);
+    if (data.volume != 0)  // volume is sqrt of product of eigenvalues
```

Comment:
I see why this change was added, but comparing floats this way is unlikely to never be false.  What about switching it to use the following above `bool zero_volume {false}` then set it to true in the if statement and check it here?

Reply:
Another option would be to just calculate the volume because the product should always be greater than zero. It would also probably be good to add this just as a precaution `assert(data.eigen_values.prod() > 0)`. You don't have to add this as part of the PR, but will not mind if you do.

Reply:
> To be honest, the assertion is not necessary either, as the eigenvalues of a matrix AA^t are the squares (and thus positive) of the (real) singular values of A.

Yea, though I have had issue in the past with this code where I was getting eigen values slightly negative which is why there is the check above. I believe the check should solve the issue but cannot be 100% sure.

Reply:
That makes sense. If that is the case then the check may no longer be needed.